### ISSUE TITLE
OrbitControls not working with custom Camera and frameloop="demand" 

### ISSUE THREAD

**Issue by biowaffeln** [pmndrs/drei#886]
- `three` version: 0.139.0
- `@react-three/fiber` version: 8.0.12
- `@react-three/drei` version: 9.6.2

### Problem description:

OrbitControls stop working when you have both a custom Camera (e.g. `<PerspectiveCamera makeDefault />`) and `frameloop="demand"` set on the Canvas.

### Relevant code:

[Example Sandbox](https://codesandbox.io/s/orbit-controls-frameloop-0digwq)

This looks like a regression from v7 of @react-three/fiber, if I downgrade to that version the above example works.

**Comment by biowaffeln** [comment]
Oh yeah, thanks for the reminder! This issue is actually a duplicate of https://github.com/pmndrs/drei/issues/873 and https://github.com/pmndrs/drei/issues/895. Classic "dependencies missing from `useEffect`" error, this pull request here https://github.com/pmndrs/drei/pull/878 should fix this issue. I'll ask in the maintainers in the discord if they can merge it.

**Comment by tytremblay** [comment]
Looks like this is going into master, which would be 9.0+, which would require (via @react-three/fiber) React 18?

**Comment by biowaffeln** [comment]
> Looks like this is going into master, which would be 9.0+, which would require (via @react-three/fiber) React 18?

Ah, didn't see you had this issue in drei v8. Yes that's correct, the fix is only for drei v9. However, I'm pretty sure the issue was introduced by this commit here https://github.com/pmndrs/drei/commit/5eff8d8cd20b9f32b6bedb67d0a33b23c70c7d58 which only landed in v9, so... maybe you have a different problem? Do you have a codesandbox that reproduces this issue with your version?

**Comment by tytremblay** [comment]
Here you go: https://codesandbox.io/s/distracted-driscoll-ztx5ge

**Comment by gpoole** [comment]
Unfortunately I had this in v8.20.2 too but reproduced in a sandbox using v9 just to make sure it was still there when I logged #895.

**Comment by biowaffeln** [comment]
Yep you're right, I misread the message on that commit. I'm not sure if they're still releasing fixes for drei v8... But the fix is pretty minimal, in the mean time you can create your own version of `OrbitControls` and add `controls` in the dependency array yourself.

**Comment by joshuaellis** [comment]
> Yep you're right, I misread the message on that commit. I'm not sure if they're still releasing fixes for drei v8... But the fix is pretty minimal, in the mean time you can create your own version of `OrbitControls` and add `controls` in the dependency array yourself.

We don't back port issue fixes unfortunately, i'm also not sure how nice this would play with `semantic-release`.

Because the issue is not apparent in the the latest version of `drei` in afraid we'd have to close this issue, see this sandbox to show its working https://codesandbox.io/s/nervous-grothendieck-3cnls5?file=/src/App.tsx

**Comment by gsimone** [comment]
I'd suggest just copy pasting the fixed component to your own codebase since it's a few lines anyway

**Comment by tytremblay** [comment]
> I'd suggest just copy pasting the fixed component to your own codebase since it's a few lines anyway

This only works if I update to React 18, since the latest version of drei, which only recently received this fix) requires React 18 via fiber.
